enhancements/installer/ibm-cloud-for-power-vs-ipi.md

```diff
@@ -0,0 +1,381 @@
+Provider][ibm-cloud-terraform-provider]. This provider will interface with IBM Cloud APIs and ultimately enable the
+OpenShift Installer to provision VM resources using Power VS, that will be used as worker and masters of the cluster.
+Unlike some of the existing providers, this depends on image availability of CoreOS in the Power VS cloud in the OVA
+format.
+
+The expected workflow would be as follows:
+1. The installer is invoked with credentials for ones IBM Cloud account.
+2. Once invoked, the installer interacts with a terraform provider to create a new Power VS service.
+3. The installer or terraform provider will then import a CoreOS OVA boot image file into the Power VS service from IBM
+Cloud Object Storage, which creates a corresponding boot image.
+4. Once imported the installer can start building the private network for the nodes.
+5. The bootstrap and master nodes will be created on in the private network. These are called Power VS instances, and
+they get created using the CoreOS boot image.
+6. In order to expose this private network and provide load balancing, the IBM Cloud APIs will leverage the IBM Cloud
+[Direct Link][ibm-direct-link-website] offering. This will allow for both the ingress load balancer and the cluster nodes to
+talk over the private network on a high-speed link.
+7. Bootstrapping completes as normal.
+8. The workers will be added using the openshift/[cluster-\*machine-\*] controllers and operators based on the openshift
+MAPI model. A Power VS controller needs to be implemented and hooked into the [OpenShift Machine
+Operator][openshift-machine-operator].
+9. The cluster teardown can make use of the existing [go client for Power][power-go-client] via `openshift-install
+destroy cluster`.
+
+### User Stories
+
+- A user of this installer would like to see a cluster running within an hour of kicking off the installer.
+- The expectation is that this is fully automated.
+
+### Implementation Details/Notes/Constraints
+
+#### Existing APIs in the Upstream
+There is a working [implementation of a machine API](https://github.com/openshift-powervs/cluster-api-provider-powervs)
+we've created for testing purposes. Though not ready for upstream, many have been tested and found to be working, such
+as mapi and nodeupdate controller. This was tested with machineset as well, so things look promising.
+
+#### Cloud Controller Manager
+We are planning to reuse the [cloud-provider-ibm][cloud-provider-ibm] for everything except for the node controller in
+machine api because the compute target doesn't support Power VS instances.
+
+#### DHCP
+The Power VS environment doesn't currently have a way of handling DHCP within the environment. We've asked their team to
+look into how they can provide this, since static IPs are currently not supported for IPI installs.
+
+The team has proposed a phased approach to DHCP support. The first stage would involve providing DHCP services by
```

Comment:
Just curious, how would this be created?

Reply:
we've got this implemented already in Power VS. it's created under the covers when you create a DHCP network in your Service Instance. Honestly we may not have needed to expose this level of detail in the enhancements proposal. And also -- the design changed a bit so taht the VM *is* hidden (and free).

Reply:
Sounds to me like DHCP is now supported, can we get things updated to reflect that?

Reply:
It's only supported for OCP, so, yes, we should rework this section completely with that in mind. I still would rather keep the detail to a minimum if that's ok.

Reply:
Yeah since there is dhcp now, I'm just going to remove that detail